src/CloudNative.CloudEvents/CloudEvent.cs

```diff
@@ -384,6 +385,26 @@ public CloudEvent Validate()
             throw new InvalidOperationException($"Missing required attributes: {joinedMissing}");
         }
 
+        /// <summary>
+        /// Validates that this CloudEvent is valid in the same way as <see cref="IsValid"/>,
+        /// but throwing an <see cref="ArgumentException"/> using the given parameter name
+        /// if the event is invalid. This is typically used within protocol bindings as the last
+        /// step in conversion.
+        /// </summary>
+        /// <param name="paramName">The parameter name to use in the exception if the event is invalid.</param>
+        /// <exception cref="ArgumentException">The event is invalid.</exception>
+        /// <returns>A reference to the same object, for simplicity of method chaining.</returns>
+        public CloudEvent ValidateForConversion(string paramName)
```

Comment:
This feels a bit odd as a public API. I understand that we don't want to duplicate this code for each protocol binding but I wonder if this would be a good candidate for using shared source instead.

Reply:
Maybe. Shared source feels like it ends up putting "official" packages (i.e. ones in this repo) on a different standing than other implementations, which can't use the same sort.

How about having this as an extension method within a namespace intended to help implementations for event formats and protocol bindings? (There are other things we could put there too.) That keeps it out of the face of regular users, but keeps a level playing field.

Reply:
Even in a separate namespace, it would still appear to users when using Intellisense if it is in the same package. It would probably at least sort it to the bottom so it might be an improvement though. I don't feel strongly here, but just wanted to call out that it seemed unusual as a public API, so good with whatever you decide.